### DISCUSSION TITLE
The format and check commands provide inconsistent handling of the order of import statements

### DISCUSSION THREAD

**Discussion by AaronConlon** [biomejs/biome#2777] [Q&A] [answered]
Hi there, I don't know how to fix it.

When I format the code:
```ts
import { fdir } from "fdir";
import ignore from "ignore";
import { existsSync, readFileSync } from "node:fs";
import { join, relative, resolve } from "node:path";
```
And if I unsafe check it which will be changed the order:
```ts
import { existsSync, readFileSync } from "node:fs";
import { join, relative, resolve } from "node:path";
import { fdir } from "fdir";
import ignore from "ignore";
```

Thanks.

**Comment by ematipico** [comment, marked as answer]
That's correct. That's how the import sorting works: https://biomejs.dev/analyzer/import-sorting/#how-imports-are-sorted

If you don't need it, you should disable it via configuration or CLI.

**Reply by AaronConlon** [reply]
Thanks.But why it is different sort rule between format and check command?

**Reply by ematipico** [reply]
Import sorting is neither a lint rule nor a formatting step.

Due to the nature of the language (JavaScript is side effect language) it can't be part of the normal formatter. However, it shouldn't be considered a lint rule because sorting is considered a different operation in the LSP.

That's why you apply import sorting using the `check` command
